quaistats/quaistats.go

```diff
@@ -828,10 +823,82 @@ func (s *Service) reportStats(conn *connWrapper) error {
 			GasPrice: gasprice,
 			Syncing:  syncing,
 			Uptime:   100,
+			Chain: common.NodeLocation.Name(),
 		},
 	}
 	report := map[string][]interface{}{
 		"emit": {"stats", stats},
 	}
 	return conn.WriteJSON(report)
 }
+
+
+// peerStats is the information to report about peers.
+type peerStats struct {
+	Chain string `json:"chain"`
+	Count int `json:"count"`
+	PeerData []*peerData `json:"peerData"`
+}
+
+// peerStat is the information to report about peers.
+type peerData struct {
+	Chain 			string `json:"chain"`
+	Enode      		string `json:"enode"` // Unique node identifier (enode URL)
+	SoftwareName    string `json:"softwareName"` // Node software version
+	LocalAddress  	string `json:"localAddress"`
+	RemoteAddress 	string `json:"remoteAddress"`
+	RTT 			string `json:"rtt"`
+	LatestHeight	uint64 `json:"latestHeight"`
+	LatestHash		string `json:"latestHash"`
+	// TODO
+	RecvLastBlockTime time.Time `json:"recvLastBlockTime"`
+	Uptime			string `json:"uptime"`
+	PeerUptime		string `json:"peerUptime"`
+}
+
+// reportPeers retrieves various stats about the peers for a node.
+func (s *Service) reportPeers(conn *connWrapper) error {
+	// Assemble the node stats and send it to the server
+	log.Trace("Sending peer details to quaistats")
+
+	peerInfo := s.backend.Downloader().PeerSet()
+
+	allPeerData := make([]*peerData, 0)
+
+	srvPeers := s.server.Peers()
+
+	for _, peer := range srvPeers {
```

Comment:
This loop populates 9 fields of `peerData`, but `peerData` has 11 fields in total. I think `upTime` and `connectedTime` are missing?

Reply:
See above.

Reply:
If we proceed with `PeerUptime` it should be a separate PR imo. Will require some deeper changes in the peer connection to pass that data.